misaligned.js

```diff
@@ -1,16 +1,31 @@
-const {expect} = require('chai')
+const { expect } = require('chai');
 
-function print_color_map() {
-    const majorColors = ["White", "Red", "Black", "Yellow", "Violet"];
-    const minorColors = ["Blue", "Orange", "Green", "Brown", "Slate"];
-    for (let i = 0; i < majorColors.length; i++) {
-        for (let j = 0; j < minorColors.length; j++) {
-            console.log(`${i * 5 + j} | ${majorColors[i]} | ${minorColors[j]}`);
-        }
-    }
-    return majorColors.length * minorColors.length;
+const majorColors = ['White', 'Red', 'Black', 'Yellow', 'Violet'];
+const minorColors = ['Blue', 'Orange', 'Green', 'Brown', 'Slate'];
+
+function print_color_map(
+	primaryColor,
+	secondaryColor,
+	printColor,
+	getColorsCount
+) {
+	printColor(primaryColor, secondaryColor);
+	let colorCount = getColorsCount(primaryColor, secondaryColor);
+	return colorCount;
+}
+
+function printColor(primaryColor, secondaryColor) {
+	for (let i = 0; i < majorColors.length; i++) {
+		for (let j = 0; j < minorColors.length; j++) {
+			console.log(`${i * 5 + j} | ${majorColors[i]} | ${minorColors[j]}`);
+		}
+	}
+}
+
+function getColorsCount(primaryColor, secondaryColor) {
+	return primaryColor.length * secondaryColor.length;
 }
 
-result = print_color_map();
+result = print_color_map(majorColors, minorColors, printColor, getColorsCount);

```

Comment:
The test is passing even when the output is wrong. To make the program testable, isolate the forming (calculation) from the printing
To test the alignment, isolate the formatting from the printing.